tests/unit/plugins/test_enricher.py

```diff
@@ -19,18 +19,132 @@
 import pytest  # NOQA
 from gordon import interfaces
 
+from gordon_gcp import exceptions
 from gordon_gcp.plugins import enricher
+from gordon_gcp.plugins import event_consumer
 
 
-def test_implements_interface():
+INSTANCE_NETWORK_DATA = {
+    'networkInterfaces': [{
+        'accessConfigs': [{
+            'natIP': '35.229.173.136'
+        }],
+    }]
+}
+
+
+@pytest.fixture
+def config(fake_keyfile):
+    return {
+        'keyfile': fake_keyfile,
+        'scopes': [],
+        'dns_zone': 'example.com',
+        'retries': 5
+    }
+
+
+def test_implements_interface(config):
     """GCEEnricher implements IEnricherClient"""
-    config = {'foo': 'bar'}
     success, error = asyncio.Queue(), asyncio.Queue()
-    client = enricher.GCEEnricher(config, success, error)
+    client = enricher.GCEEnricher(config, None, success, error)
 
     assert interfaces.IEnricherClient.providedBy(client)
     assert interfaces.IEnricherClient.implementedBy(enricher.GCEEnricher)
     assert config is client.config
     assert success is client.success_channel
     assert error is client.error_channel
     assert 'enrich' == client.phase
+
+
+@pytest.fixture
+def event_msg(mocker):
+    fake_data = {
+            'action': 'additions',
+            'resourceName': 'projects/project1/zones/zone1/instances/instance1',
+            'resourceRecords': [],
+            'timestamp': '2017-12-04T20:13:44.181Z',
+        }
+    pubsub_msg = mocker.Mock()
+    pubsub_msg._ack_id = '1:1'
+    return event_consumer.GEventMessage(
+        pubsub_msg, fake_data, enricher.GCEEnricher.phase)
+
+
+@pytest.fixture
+def channel_pair():
+    return asyncio.Queue(), asyncio.Queue()
+
+
+@pytest.fixture
+def mock_http_client(mocker, get_mock_coro):
+    get_json_mock, get_json_coro = get_mock_coro()
+    http_client = mocker.Mock()
+    mocker.patch.object(http_client, 'get_json', get_json_coro)
+    mocker.patch.object(http_client, '_get_json', get_json_mock)
+    return http_client
+
+
+@pytest.fixture
+def mock_async_sleep(mocker, get_mock_coro):
+    sleep_mock, sleep_coro = get_mock_coro()
+    mocker.patch('gordon_gcp.plugins.enricher.asyncio.sleep', sleep_coro)
+    return sleep_mock
+
+
+@pytest.mark.parametrize('network_data,sleep_calls,logs_logged', [
+    ([INSTANCE_NETWORK_DATA], 0, 3),
+    ([{}, {}, INSTANCE_NETWORK_DATA], 2, 5)])
+@pytest.mark.asyncio
+async def test_process_event_msg(config, event_msg, channel_pair,
+                                 mock_async_sleep, mock_http_client, caplog,
+                                 network_data, sleep_calls, logs_logged):
+    success_chnl, error_chnl = channel_pair
+    mock_http_client._get_json.side_effect = network_data
+    gce_enricher = enricher.GCEEnricher(config, mock_http_client, success_chnl,
+                                        error_chnl)
+
+    await gce_enricher.process(event_msg)
+
+    event_message = await success_chnl.get()
```

Comment:
Naming is a bit confusing with `event_msg` fixture. Maybe something like `ret_event_msg`? just something to clearly differentiate.

Reply:
ಠಿ_ಠಿ, good point!